docs/typescript.mdx

```diff
@@ -248,34 +248,33 @@ const App = () => (
 ### Define a Theme
```

Comment:
The whole docs on Theme usage with TS feels massively outdated now, and I feel like it should be reworded altogether.

_Exposing for discussion._

Reply:
Right - I feel like we can handle it in a separate PR though. Ideally, such docs would present the current solution with its tradeoffs and incorporate stuff from this https://github.com/emotion-js/emotion/pull/973

Reply:
Yes, this is crucial and I plan to get this done before v11 👍